_posts/2024-07-04-cleanup-rawhide-builds.md

```diff
@@ -0,0 +1,59 @@
+---
+title: Rawhide (aka "rolling") chroot EOL policy in Copr
+author: Pavel Raiskup
+layout: post
+---
+
+In Fedora Copr, we strive to avoid wasting disk storage unnecessarily.  Disk
+storage is the commodity that incurs the highest cloud costs.  For example, we
+clean the build results for EOL chroots, we clean source builds (build results
+that are imported into DistGit), we clean the DistGit lookaside cache, build
+logs, uploaded files, failed build logs, and more.  Yet, we maintain >= 40TB of
+data with RAID redundancy, volume snapshots, ...
+
+[Most of the Fedora Copr backend
+storage](https://copr-be.cloud.fedoraproject.org/stats/distro.html) is used for
+holding the build results related to Fedora, with the largest consumer being the
+Fedora Rawhide chroot.
+
+The "problem" with Rawhide chroots is that these are "rolling" ones, which never
+reach EOL (unlike the branched Fedoras; see, for example, the data consumption
+for the old Fedora 35 in the previous link if you want to see the difference).
+
+Yet, _most_ existing projects _do_ build for Rawhide -- and until now, we did
+not have a tool for performing any cleanups there.  Now we do.
+
+**Warning**: **Fedora Rawhide** is not the only target distro suffering from
+this problem. There are other rolling distros we build for, where we'll apply
+the same policy, namely **Fedora ELN**, **openSUSE Tumbleweed**, and **Mageia
+Cauldron**.
+
+## The Rolling Policy Being Implemented
+
+In [this Copr issue](https://copr-be.cloud.fedoraproject.org/stats/distro.html),
+we concluded that a viable option is to detect the liveness of particular
+"rolling" chroots (enabled in specific Copr projects).  If a chroot appears
```

Comment:
> to detect the liveness of particular "rolling" chroots (enabled in specific Copr projects).

I would go with 'to detect the liveness of particular "rolling" chroots in Copr projects' to avoid the text in brackets.

Reply:
Updated, good idea.